youtube-scripts.js

```diff
@@ -3035,13 +3037,17 @@ ImprovedTube.videoPageUpdate = function() {
 ------------------------------------------------------------------------------*/
 
 ImprovedTube.video_src = false;
+ImprovedTube.initialVideoUpdateDone = false;
 
 ImprovedTube.videoUpdated = function() {
     this.playerAds();
     this.playerPlaybackSpeed();
-    this.playerQuality();
     this.playerVolume();
     this.playlistUpNextAutoplay();
+
+    if (ImprovedTube.initialVideoUpdateDone !== true) {
+        this.playerQuality();
```

Comment:
According to comments in #813 all of these are affected and will be re-enforced on quality change:
```suggestion
        this.playerPlaybackSpeed();
        this.playerQuality();
        this.playerVolume();
        this.playlistUpNextAutoplay();
```
Not sure if #808 will still be fixed by this, though, as people say they reset after the first ad. Kinda contradictory to #813/#828.
I wasn't paying attention to quality enforcement after ads myself, so I can't confirm this.

Reply:
Oh very good point. I hadn't considered ads, and I do all of my testing with an adblocker. Let me try and see!
Honestly if ads reset the video quality, I'm not quite sure what to do. I suppose I'll have to add a listener for when an ad stops playing..

Reply:
Ok.. I just spent a while implementing a fix that resets the video quality, etc. after an ad is played, and then realized that this probably isn't even desired functionality. 🤦  It probably was fine already, since if you've manually set the quality etc. you're fine with it staying that way after the video. Durr. Okay, I'll revert my changes.